graphql/util.go

```diff
@@ -956,3 +928,88 @@ func getProjectMetadata(ctx context.Context, projectId *string, patchId *string)
 	}
 	return &apiProjectRef, nil
 }
+
+//////////////////////////////////////////
+// Helper functions for task test results.
+//////////////////////////////////////////
+
+func convertTestFilterOptions(ctx context.Context, dbTask *task.Task, opts *TestFilterOptions) (*testresult.FilterOptions, error) {
+	if opts == nil {
+		return nil, nil
+	}
+
+	sortBy, sortOrderDSC, baseTaskOpts, err := convertTestSortOptions(ctx, dbTask, opts.Sort)
+	if err != nil {
+		return nil, err
+	}
+
+	return &testresult.FilterOptions{
+		TestName:     opts.TestName,
+		Statuses:     opts.Statuses,
+		GroupID:      opts.GroupID,
+		SortBy:       sortBy,
+		SortOrderDSC: sortOrderDSC,
+		Limit:        opts.Limit,
+		Page:         opts.Page,
+		BaseTasks:    baseTaskOpts,
```

Comment:
nil handling could be handled here for the missing values.

Reply:
Done!